python/tk_multi_loader/model_item_data.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk import TankError
+from sgtk.platform.qt import QtCore
+
+shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
+
+
+def get_item_data(item):
```

Comment:
All the Loader code in based on (expects) certain `SG_DATA_ROLE` and `SG_ASSOCIATED_FIELD_ROLE` data formats retrieved from `ShotgunModel` items.

For `ShotgunHierarchyModel` items, `SG_DATA_ROLE` data has a very different format, and there is no `SG_ASSOCIATED_FIELD_ROLE` data.

Centralizing the _standardization_ of the item data in this function was the cleaner approach with the current `ShotgunModel` and `ShotgunHierarchyModel` item data.